include/db_connector

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# Prowler - the handy cloud security tool (copyright 2018) by Toni de la Fuente
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+# DB connectors
+# Supported connectors list is stored in SUPPORTED_DB_PROVIDERS
+# Once the logic of the connector is added it is needed to add it to the list also
+# Example: adding oracle to the list
+# SUPPORTED_DB_PROVIDERS="postgresql|oracle"
+# oracle_connector() { ... }
+
+SUPPORTED_DB_PROVIDERS="postgresql"
+
+# Connectors for different databases
+#POSTGRESQL
+postgresql_connector () {
+
+    PSQL_USER=$(awk -F ':' '{print $4}' ~/.pgpass)
+    PSQL_HOSTNAME=$(awk -F ':' '{print $1}' ~/.pgpass)
+    PSQL_PORT=$(awk -F ':' '{print $2}' ~/.pgpass)
+    CSV_REGISTRY="${1}"
+
+    # Previous checks after sending data to db
+    # psql is installed and in path
+    if ! command -v "psql" > /dev/null 2>&1
+    then
+        db_exit_abnormally "postgresql" "psql tool not installed or not found"
+    # database is reachable            
+    elif ! pg_isready -q  -U "${PSQL_USER}" -h "${PSQL_HOSTNAME}" -p "${PSQL_PORT}"
+    then
+        db_exit_abnormally "postgresql" "Database listening on host ${PSQL_HOSTNAME} on port ${PSQL_PORT} connected with user ${PSQL_USER} is unreachable"     
+    fi
```

Comment:
I think it's better to test this once prior to start execution. What do you think?

Reply:
Added prior execution, review it